rc/windowing/kitty.kak

```diff
@@ -1,8 +1,15 @@
+# https://sw.kovidgoyal.net/kitty/index.html
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
 
-hook -group kitty-hooks global KakBegin .* %sh{
-    if [ "$TERM" = "xterm-kitty" ] && [ -z "$TMUX" ]; then
-        echo "require-module kitty"
-    fi
+define-command -hidden detect-module-kitty -docstring '
+detect-module-kitty: determine if the kitty module is relevant to the current
+environment' \
+%{
+    evaluate-commands %sh{
+        if [ "$TERM" = "xterm-kitty" ]; then
```

Comment:
I think the following is a better heuristic here, might as well fix it:

```suggestion
        if [ -n "${KITTY_WINDOW_ID}" ]; then
```

Reply:
Either version actually works, but I think the $TERM value is more "correct", since kitty explicitly documents that $TERM must be set to "xterm-kitty" otherwise things may break

Reply:
Some values for this variable are known to be interchangeable.

That's why `tmux.kak` checks the `$TMUX` environment variable (same with `screen.kak` and `$STY`), and not if `$TERM = tmux-256color` - because you can also use `screen-256color` simultaneously for Tmux, Screen and Kitty.

In fact, when debugging Kakoune rendering issues with users, we often ask them to test with `TERM=screen-256color`.